### PR TITLE
Blog submission for ptp-notification

### PR DESCRIPTION
**Pull request by Fanta-astic**
Submitting a blog post for:
ORAN Notification for Monitoring System Timing Synchronization

Pertains to stx 9.0 onwards.

### CODE REVIEW

File: src/pages/blog/starlingx-oran-notification-for-ptp.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+itself: how can these workloads determine if the timing state healthy and accurate?
+
+On a single StarlingX AIO-SX system, there may be multiple ptp4l and ts2phc instances writing time
+to various PHCs, a phc2sys instance keeping the realtime system clock in sync, multiple GNSS inputs
+and several PPS connections between NICs. Any degradation to one of these components can have
+significant impacts to the overall precision of the system timing and may require user workloads to
+act in response. For example, a 5G radio unit may be configured to stop transmitting until time
+synchronization is restored.
+
+Available in the 9.0 StarlingX release, the ptp-notification application to enables user workloads
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-notification-for-ptp.md:23]:
```suggestion
Available starting from the 9.0 StarlingX release, the ptp-notification application to enables user workloads
```

File: src/pages/blog/starlingx-oran-notification-for-ptp.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+review the major functionality provided by ptp-notification.
+
+The full standard can be downloaded from the [O-RAN Alliance
+website](https://orandownloadsweb.azurewebsites.net/specifications).
+
+
+# The ptp-notification application
+The ptp-notification application is a system-managed application that is packaged with StarlingX. It
+can be deployed and configured using the various "system application-*" commands. Installing
+ptp-notification is simple process of labelling the specific hosts that should run the application,
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-notification-for-ptp.md:38]:
```suggestion
ptp-notification is a simple process of labelling the specific hosts that should run the application,
```

File: src/pages/blog/starlingx-oran-notification-for-ptp.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+uploading the application tarball, specifying a few configuration values and then deploying the
+application. Complete steps for this procedure can be found in the [documentation](https://docs.
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-notification-for-ptp.md:40]:
```suggestion
application. Complete steps for this procedure can be found in the [StarlingX PTP documentation](https://docs.
```

File: src/pages/blog/starlingx-oran-notification-for-ptp.md [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+starlingx.io/admintasks/kubernetes/install-ptp-notifications.html)
+
+The application consists of three logical functions for handling the tracking of timing state, the
+delivery of push notifications and the registration of subscribers for notifications.
+
+![alt text](/img/ptp-notification-functions.png)
+
+
+# The ptp-notification API
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-notification-for-ptp.md:49]:
```suggestion
## The ptp-notification API
```

File: src/pages/blog/starlingx-oran-notification-for-ptp.md [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+Ptp-notification provides a REST API for user applications to interact with. In the StarlingX
+documentation, it is referred to as the "v2 API", which implements the O-RAN standard. The v1 API
+predates the O-RAN standard and is not recommended for use in new deployments.
+
+The v2 API documentation is available
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-notification-for-ptp.md:54]:
```suggestion
Check out the [v2 API](https://docs.starlingx.io/api-ref/ptp-notification-armada-app/api_ptp_notifications_definition_v2.html) documentation for more details.
```

File: src/pages/blog/starlingx-oran-notification-for-ptp.md [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+            }
+         ]
+      }
+   }
+}
+```
+
+### Create and manage subscriptions
+
+User applications can be configured to subscribe to and receive push notification from the
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-notification-for-ptp.md:108]:
```suggestion
User applications can be configured to subscribe to and receive push notifications from the
```

File: src/pages/blog/starlingx-oran-notification-for-ptp.md [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+         }
+      ]
+   }
+}
+```
+
+
+# The ptp-notification sidecar
+
+In order to simplify use of the ptp-notification API, StarlingX provides the "notificationclient"
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-notification-for-ptp.md:160]:
```suggestion
In order to simplify the use of the ptp-notification API, StarlingX provides the "notificationclient"
```

File: src/pages/blog/starlingx-oran-notification-for-ptp.md [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+container image which can be deployed as a sidecar alongside user applications. User applications
+can then make their API requests via the notification client. The notification client container
+handles the process of locating the appropriate notification server and tracking active
+subscriptions for the user application.
+
+An example of integrating the notification client as a sidecar is available in the StarlingX
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-notification-for-ptp.md:166]:
```suggestion
An example of integrating the notification client as a sidecar is available in the [related StarlingX documentation](https://docs.starlingx.io/guest_integration/kubernetes/integrate-application-with-notification-client-sidecar.html).
```

File: src/pages/blog/starlingx-oran-notification-for-ptp.md [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+documentation
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-notification-for-ptp.md:167]:
```suggestion
```

File: src/pages/blog/starlingx-oran-notification-for-ptp.md [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+[here](https://docs.starlingx.io/guest_integration/kubernetes/integrate-application-with-notification-client-sidecar.html).
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-notification-for-ptp.md:168]:
```suggestion
```

File: src/pages/blog/starlingx-oran-notification-for-ptp.md [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+enables timing sensitive applications to take appropriate steps when timing accuracy degrades.
+
+The StarlingX community is continuously working on enhancements to the PTP support in the platform. 
+Stay tuned for an upcoming blog post describing recent work on a High Availability solution for
+phc2sys, which allows multiple clock sources to synchronize the system clock.
+
+If you would like to learn more about the project and get involved check the 
+[website](https://www.starlingx.io) for more information or 
+[download the code](https://opendev.org/starlingx) and start to experiment with the platform.
+
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-notification-for-ptp.md:181]:
```suggestion
If you are already evaluating or using the software please fill out the [user survey](https://openinfrafoundation.formstack.com/forms/starlingx_user_survey) and help the community improve the project based on your feedback.
```

**Review by ildikov — COMMENTED** [review]
Hi Cole,

Thank you for submitting this PTP article for the blog.  I added a few minor suggestions in line, otherwise it looks great!

I scheduled it for a bit over a month from now, as there's another post to go live soon and I'm spacing out the remaining ones. I will move this up if I got more content in the meantime.

Best Regards,
Ildikó

**Comment by Fanta-astic** [comment]
Thanks @ildikov! I have applied your suggested changes.

**Pull request merged**